2 - Counter Quantity [#2dmz47h]

isa9x opened the pull request:
Penambahan widget content quantity

cara pakai:
![image](https://user-images.githubusercontent.com/10303178/158012439-3bee5fc7-991d-4c40-aefc-b80c1e7e7b2f.png)

Screenshot Potrait:
![image](https://user-images.githubusercontent.com/10303178/158012463-cabb3ad4-47bd-46d5-a29f-dae110149b11.png)

Screenshot Landscape:
![image](https://user-images.githubusercontent.com/10303178/158012476-d0cb69b1-30e6-4c11-99d5-60687953bbb5.png)

isa9x committed:
Detail AppBar

Appbar halaman detail / 2

isa9x committed:
2 Image/Slider

tambahan image slider

isa9x committed:
2 - Text content

penambahan text content di halaman detail

isa9x committed:
2 - Counter Quantity

penambahan counter pada detail

vstacked commented on lib/page/detail/widget/detail_counter.dart:
```diff
@@ -0,0 +1,115 @@
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        InkWell(
+          onTap: decrease,
+          child: Container(
+            decoration: BoxDecoration(
+                gradient: linearGradient,
+                borderRadius: BorderRadius.circular(10)),
+            height: widget.screenHeight * 0.040,
+            width: widget.screenHeight * 0.040,
+            child: const Icon(
+              Icons.remove,
+              color: Colors.white,
+            ),
+          ),
+        ),
```
dibikin internal widget mungkin mas? buat yang decrease sama increase biar ngga duplikasi

nfach98 reviewed (approved):
Good

StanislausPN commented on lib/page/detail/widget/detail_counter.dart:
```diff
@@ -0,0 +1,115 @@
+          child: SizedBox(
+            height: widget.screenHeight * 0.040,
+            width: widget.screenHeight * 0.040,
+            child: TextField(
```
Mau tanya mas, kenapa dibuat textfield?

isa9x replied:
yang pertama terlintas pakai Textfield saya mas, mungkin ada saran yang lebih baik mas?

StanislausPN commented on lib/page/detail/widget/detail_content.dart:
```diff
@@ -0,0 +1,66 @@
+      children: [
+        Padding(
+          padding: EdgeInsets.only(
+              top: screenHeight * 0.05,
```
menurutku untuk padding dan height width, jangan dikasih media query mas, mending di paten saja sesuai ukuran di figma. Dulu pernah pengalaman dibikin seperti ini, ketika dia ambil layar besar, ukurannya jadi memanjang.

isa9x committed:
pembuatan widget untuk button

pembuatan widget untuk button agar tidak redudant

mnafian commented on lib/page/detail/widget/detail_slider.dart:
```diff
@@ -0,0 +1,129 @@
+                  return GestureDetector(
+                    onTap: () => _controller.animateToPage(entry.key),
+                    child: _current == entry.key
+                        ? Container(
```
ini di extract jadi function widget diluar build method biar gak terlalu kompleks ternary operatornya

isa9x replied:
ini saya perbaiki di branch images_slider juga ya mas, 
![image](https://user-images.githubusercontent.com/10303178/158166022-2a99de48-f3f4-4ae1-8142-1979095bbcb6.png)

mnafian commented on lib/page/detail/widget/detail_slider.dart:
```diff
@@ -0,0 +1,129 @@
+                  return GestureDetector(
+                    onTap: () => _controller.animateToPage(entry.key),
+                    child: _current == entry.key
+                        ? Container(
```
ini di extract jadi function widget diluar build method biar gak terlalu kompleks ternary operatornya

isa9x committed:
perbaikan sesuai comment

1.untuk  slider, widget yang di ternary nya sudah di extract
2. perbaikan di konten, paddingnya mengikuti figma
3. untuk di counter, text field diganti container,

isa9x committed:
Penggantian mediaquery

media query diganti dengan ukuran patokan figma

nfach98 committed:
Merge branch 'main' into 2_counter_quantity
